clang/lib/CIR/CodeGen/CIRGenExpr.cpp

```diff
@@ -653,9 +653,62 @@ RValue CIRGenFunction::buildLoadOfLValue(LValue LV, SourceLocation Loc) {
         getLoc(Loc), load, LV.getVectorIdx()));
   }
 
+  if (LV.isExtVectorElt()) {
+    return buildLoadOfExtVectorElementLValue(LV);
+  }
+
   llvm_unreachable("NYI");
 }
 
+int64_t CIRGenFunction::getAccessedFieldNo(unsigned int idx,
+                                           const mlir::Attribute elts) {
+  assert(elts.isa<mlir::ArrayAttr>());
+
+  auto elt = elts.cast<mlir::ArrayAttr>()[idx];
+  assert(elt.isa<mlir::IntegerAttr>());
```

Comment:
If you are doing a `isa` followed by `cast`, better just do `dyn_cast` and assert if it's null.